### PR TITLE
Partyrole implemented

### PR DESCRIPTION
**Pull request by sluFicodes**

### CODE REVIEW

File: party-role/src/main/java/org/fiware/tmforum/partyrole/domain/AgreementRef.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package org.fiware.tmforum.partyrole.domain;
+
+import java.util.List;
+
+import org.fiware.tmforum.common.domain.RefEntity;
+
+import lombok.EqualsAndHashCode;
+
```
Comment (wistefan) [inline party-role/src/main/java/org/fiware/tmforum/partyrole/domain/AgreementRef.java:8]: Whats the reason to not use the AgreementRef from the commons module?

File: party-role/src/main/java/org/fiware/tmforum/partyrole/domain/AccountRef.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.fiware.tmforum.common.domain.RefEntity;
+
+import io.github.wistefan.mapping.annotations.AttributeGetter;
+import io.github.wistefan.mapping.annotations.AttributeSetter;
+import io.github.wistefan.mapping.annotations.AttributeType;
+import io.github.wistefan.mapping.annotations.Ignore;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+
```
Comment (wistefan) [inline party-role/src/main/java/org/fiware/tmforum/partyrole/domain/AccountRef.java:15]: AccountRef, CreditProfile and PaymentMethodRef already exist as part of the customer-management module. It should be moved to  a shared module and reused

File: party-role/src/main/java/org/fiware/tmforum/partyrole/rest/PartyRoleController.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+     private Mono<PartyRole> getCheckingMono(PartyRole pr) {
+                List<List<? extends ReferencedEntity>> references = new ArrayList<>();
+                references.add(pr.getAccount());
+                references.add(pr.getAgreement());
+                references.add(pr.getPaymentMethod());
+                references.add(pr.getRelatedParty());
+                Optional.ofNullable(pr.getEngagedParty()).map(List::of).ifPresent(references::add);
+                return getCheckingMono(pr, references)
+                                .onErrorMap(throwable -> new TmForumException(
+                                                String.format("Was not able to create individual %s",
```
Comment (wistefan) [inline party-role/src/main/java/org/fiware/tmforum/partyrole/rest/PartyRoleController.java:88]: Exception Message should be updated.

File: party-role/src/main/java/org/fiware/tmforum/partyrole/rest/PartyRoleController.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        PartyRole pr = tmForumMapper.map(partyRoleUpdateVO, id);
+
+        return patch(id, pr, getCheckingMono(pr), PartyRole.class)
+                        .map(tmForumMapper::map)
+                        .map(HttpResponse::ok);    
+    }
+    @Override
+    public Mono<HttpResponse<PartyRoleVO>> retrievePartyRole(@NonNull String id, @Nullable String fields) {
+        return retrieve(id, PartyRole.class).switchIfEmpty(
+             Mono.error(new TmForumException("No such individual exists.", TmForumExceptionReason.NOT_FOUND)))
```
Comment (wistefan) [inline party-role/src/main/java/org/fiware/tmforum/partyrole/rest/PartyRoleController.java:74]: Exception Message should be updated.

File: party-role/src/main/java/org/fiware/tmforum/partyrole/rest/PartyRoleController.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        .map(prStream -> prStream.map(tmForumMapper::map).toList())
+        .switchIfEmpty(Mono.just(List.of()))
+        .map(HttpResponse::ok);
+return res;    
+    }
+    @Override
+    public Mono<HttpResponse<PartyRoleVO>> patchPartyRole(@NonNull String id, @NonNull PartyRoleUpdateVO partyRoleUpdateVO) {
+        // non-ngsi-ld ids cannot exist.
+        if (!IdHelper.isNgsiLdId(id)) {
+                throw new TmForumException("Did not receive a valid id, such agreement cannot exist.",
```
Comment (wistefan) [inline party-role/src/main/java/org/fiware/tmforum/partyrole/rest/PartyRoleController.java:62]: Exception Message should be updated.

**Review by wistefan — CHANGES_REQUESTED** [review]

**Comment by sluFicodes** [comment]
Hi, all changes have been updated.

**Comment by wistefan** [comment]
@sluFicodes Francisco mentioned that you need to run TMForum against Scorpio, however your tests fail with Scorpio. Can you please check? Just run "mvn clean test integration-test verify -Dbroker=scorpio" to test with it.

File: party-role/src/main/resources/application.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
+micronaut:
+  application:
+    name: ${project.artifactId}
+
+  caches:
```
Comment (beknazaresenbek) [inline party-role/src/main/resources/application.yaml:5]: There were some changes in application properties. Now there are multiple properties for different environments. Could you adjust it? (As example, you can check account-module)

File: party-role/src/test/resources/application.yaml [hunk 7]
```diff
@@ -0,0 +1,45 @@
+micronaut:
+
+  server:
+    port: 8632
+
+  caches:
```
Comment (beknazaresenbek) [inline party-role/src/test/resources/application.yaml:6]: The same changes in application properties should be done for tests (check account-module, for example)

File: party-role/pom.xml [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+        </dependency>
+    </dependencies>
+
+    <build>
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
+                <filtering>true</filtering>
+            </resource>
+            <resource>
```
Comment (beknazaresenbek) [inline party-role/pom.xml:195]: Could you put test resources in testResources tag? (check agreement-module, for example)

**Review by beknazaresenbek — CHANGES_REQUESTED** [review]

**Comment by sluFicodes** [comment]
Okay it should be fine now

**Review by beknazaresenbek — APPROVED** [review]

**Review by wistefan — APPROVED** [review]

**Pull request merged**